backfill.py

```diff
@@ -130,27 +148,126 @@ def get_result_from_durations(start_up_durations_path, analyzed_path):
     analyze_durations.save_output(stats, analyzed_path)
 
 
-def run_performance_analysis_on_nightly(package_id, path_to_measure_start_up_script, array_of_apk_path):
-    for apk_path in array_of_apk_path:
-        analyze_nightly_for_one_build(package_id, path_to_measure_start_up_script, apk_path)
+def run_performance_analysis_on_nightly(package_id, path_to_measure_start_up_script, array_of_apk_path, build_type):
+    for idx, apk_path in enumerate(array_of_apk_path):
+        analyze_nightly_for_one_build(idx, package_id, path_to_measure_start_up_script, apk_path, build_type)
+
+
+def fetch_repository(repository_path, remote_name):
+    remote_repo_name = "upstream" if len(remote_name) == 0 else remote_name
+
+    fetch_proc = subprocess.run(["git", "fetch", remote_repo_name], cwd=repository_path, capture_output=True)
+
+    if fetch_proc.returncode != 0:
+        print(("\n\nSomething went wrong while fetching this repostirory: {repo} . The associated error message was:"
+               "\n\n {error}".format(repo=repository_path, error=fetch_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+
+def get_all_commits_in_commits_range(start_commit, end_commit, repository_path):
+    commit_proc = subprocess.run(
+        ["git", "rev-list", "--ancestry-path", start_commit + "^.." + end_commit],
+        cwd=repository_path, capture_output=True, text=True)
+
+    if commit_proc.returncode != 0:
```

Comment:
issue (non-blocking): if this command fails, we're executing on a questionable input

suggestion: I feel like we should stop the script if this command fails (maybe we can use `check=True` and only capture stdout so stderr goes right to the console.